client/client.go

```diff
@@ -353,30 +353,44 @@ func (c *Client) Do(req *http.Request) (*container.Container, *http.Response, er
 	if !c.skipLoggingPayload {
 		log.Printf("[TRACE] HTTP Request Body: %v", req.Body)
 	}
-	resp, err := c.httpClient.Do(req)
-	if err != nil {
-		return nil, nil, err
-	}
-	log.Printf("[DEBUG] HTTP Request: %s %s", req.Method, req.URL.String())
-	log.Printf("[DEBUG] HTTP Response: %d %s %v", resp.StatusCode, resp.Status, resp)
-
-	bodyBytes, err := ioutil.ReadAll(resp.Body)
-	bodyStr := string(bodyBytes)
-	resp.Body.Close()
-	log.Printf("[DEBUG] HTTP response unique string %s %s %s", req.Method, req.URL.String(), bodyStr)
-	if req.Method != "DELETE" && resp.StatusCode != 204 {
-		obj, err := container.ParseJSON(bodyBytes)
 
+	// Attempts are used to retry the request in case of a JSON parse failure
+	for attempts := 1; ; attempts++ {
+		log.Printf("[DEBUG] HTTP Request: %s %s", req.Method, req.URL.String())
+		resp, err := c.httpClient.Do(req)
 		if err != nil {
-			log.Printf("Error occured while json parsing %+v", err)
+			log.Printf("[ERROR] Error occured on HTTP Request: %s", err)
+			return nil, nil, err
+		}
+		log.Printf("[DEBUG] HTTP Response: %d %s %v", resp.StatusCode, resp.Status, resp)
+
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		bodyStr := string(bodyBytes)
+		resp.Body.Close()
+		log.Printf("[DEBUG] HTTP response unique string %s %s %s", req.Method, req.URL.String(), bodyStr)
+		if req.Method != "DELETE" && resp.StatusCode != 204 {
+			obj, err := container.ParseJSON(bodyBytes)
+			if err != nil {
+				// With large schemas errors are sometimes noticed for JSON parsing
+				// Recreate is not consistent accross attempt, where some schemas that failed are parsed correctly next run
+				// According to CX investigation is related to retransmissions and lost packets
+				// Workaround for https://github.com/CiscoDevNet/terraform-provider-mso/issues/338
+				if ok := attempts <= 3; ok {
```

Comment:
Shall we add Exponential backoff-> The delay increases exponentially after each attempt?

Reply:
this is pending discussion from team